part-ii-domain-logic/deadlines.md

```diff
@@ -0,0 +1,74 @@
+# Deadlines
+
+The 'Deadline concept' in the Axon Framework is a mechanism which enables certain actions (in our case a `@DeadlineHandler` annotated method) to be executed after certain amount of time. The context of this execution is an Aggregate or a Saga in which the Deadline was scheduled. If the Deadline becomes obsolete there is the possibility to cancel it as well.  
+
+Deadlines are possible to be scheduled from Sagas and Aggregates. The `DeadlineManager` component is responsible for scheduling deadlines and invoking `@DeadlineHandler`s when the deadline is met. The `DeadlineManager` can be injected as a resource. It has two flavors: `SimpleDeadlineManager` and `QuartzDeadlineManager`, just like the [Event Scheduling](sagas.md#keeping-track-of-deadlines) mechanism for Sagas. 
+
+## Scheduling a Deadline
+
+A deadline can be scheduled only by providing a `Duration` after which it will be triggered (or `Instance` at which it will be triggered) and a name.
+
+> **Note**
+>  
+> Unlike [Event Scheduling](sagas.md#keeping-track-of-deadlines), when a Deadline is triggered there will be no storing of the published Message.
+
+```java
+String deadlineId = deadlineManager.schedule(Duration.ofMillis(500), "myDeadline");
+```
+
+As a result we receive a `deadlineId` which can be used to cancel the deadline. Cancelling a deadline could for example come in handy when a certain event means that the previously scheduled deadline has become obsolete (e.g. there is a deadline for paying the invoice, but the client payed the amount which means that the deadline is obsolete and can be canceled).
+
+```java
+deadlineManager.cancelSchedule("myDeadline", deadlineId);
+```
+
+> **Note**
+>
+> It is possible to cancel all deadlines of a given name by invoking `deadlineManager.cancelAll("myDeadline")`.
+
+If you need some contextual data about the Deadline during the Deadline Handling, you can attach a Deadline Payload when scheduling a Deadline:
+
+```java
+String deadlineId = deadlineManager.schedule(Duration.ofMillis(500), "myDeadline", new MyDeadlinePayload(...));
```

Comment:
I foresee some questions about best practices in deadline scheduling in combination with event sourcing. Where/when to store the deadlineId, for example?
Let's beat the questions and describe it.